galaxy_importer/schema.py

```diff
@@ -305,22 +305,72 @@ def _check_license_or_license_file(self):
             f"'license_file' ({self.license_file}) were invalid.")
 
 
+@attr.s(frozen=True)
+class CollectionArtifactFile(object):
+    name = attr.ib()
+    ftype = attr.ib()
+    src_name = attr.ib(default=None)
+    chksum_type = attr.ib(default="sha256")
+    chksum_sha256 = attr.ib(default=None)
+    format = attr.ib(default=1)
+
+
 @attr.s(frozen=True)
 class CollectionArtifactManifest(object):
     """Represents collection manifest metadata."""
 
     collection_info = attr.ib(type=CollectionInfo)
     format = attr.ib(default=1)
-    file_manifest_file = attr.ib(factory=dict)
+
+    file_manifest_file = attr.ib(default=None,
+                                 type=CollectionArtifactFile,
+                                 validator=attr.validators.optional(
+                                     attr.validators.instance_of(CollectionArtifactFile)))
 
     @classmethod
     def parse(cls, data):
         meta = json.loads(data)
         col_info = meta.pop('collection_info', None)
         meta['collection_info'] = CollectionInfo(**col_info)
+
+        # FIXME: Decide where we want missing info for FILES.json to raise an exception
+        file_manifest_file = meta.pop('file_manifest_file', None)
+        meta['file_manifest_file'] = CollectionArtifactFile(**file_manifest_file)
         return cls(**meta)
 
 
+def convert_list_to_artifact_file_list(val):
+    '''Convert a list of dicts with file info into list of CollectionArtifactFile'''
+
+    new_list = []
+    for file_item in val:
+        if isinstance(file_item, CollectionArtifactFile):
+            new_list.append(file_item)
+        else:
+            artifact_file = CollectionArtifactFile(name=file_item['name'],
+                                                   ftype=file_item['ftype'],
+                                                   chksum_type=file_item.get('chksum_type'),
+                                                   chksum_sha256=file_item.get('chksum_sha256'))
+            new_list.append(artifact_file)
+
+    return new_list
+
+
+# FIXME: where do we want to load / validate this.
```

Comment:
I see 6 FIXME and TODOs... they may be all valid as placeholder notes, but I doubt they will be addressed anytime soon after merge, anything we want to address before merge?

Reply:
The FIXMEs here are mostly about where/when/how we want to catch any exceptions raised and provide useful error messages,

# FIXME: Decide where we want missing info for FILES.json to raise an exception 
in CollectionArtifactManifest.parse() 

I don't think there is a unit test for a MANIFEST.json that doesn't have a file_manifest_file item in it's top level dict. That would cause a key error there that would bubble up... to at least CollectionLoader._load_collection_artifact().

I'll see if I can add a unit for _load_collection_artifact() that will trigger that case.


# FIXME: where do we want to load / validate this.
We do that in _load_collection_file_manifest() so I'll remove the FIXME

Reply:
Removed `# FIXME: where do we want to load / validate this.`